Test/Framework/Preprocessor.hs

```diff
@@ -396,8 +396,8 @@ cpphsOptions =
       boolopts = (boolopts defaultCpphsOptions) { lang = True } -- lex as haskell
     }
 
-transform :: Bool -> Bool -> FilePath -> String -> IO String
-transform hunitBackwardsCompat debug originalFileName input =
+transform :: Bool -> Bool -> Bool -> FilePath -> String -> IO String
```

Comment:
It would be good to wrap these three Bools into a datatype. Otherwise it's too likely to confuse the three options.